### PR TITLE
Use slow lane to do global resync

### PR DESCRIPTION
**Pull request by vagababov**
- This just redirects the requests to the slow queue

/assign mattmoor

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -232,6 +238,19 @@ func (c *Impl) EnqueueAfter(obj interface{}, after time.Duration) {
...
+// and enqueues it in the slow lane.
+func (c *Impl) EnqueueSlow(obj interface{}) {
+	object, err := kmeta.DeletionHandlingAccessor(obj)
+	if err != nil {
+		c.logger.Errorw("EnqueueSlow", zap.Error(err))
+		return
+	}
+	key := types.NamespacedName{Namespace: object.GetNamespace(), Name: object.GetName()}
+	c.workQueue.SlowLane().Add(key)
+	c.logger.Debugf("Adding to the slow queue %s (depth: %d)", safeKey(key), c.workQueue.Len())
```
Comment (mattmoor) [inline controller/controller.go:251]: Split off EnqueueSlowKey

File: controller/controller.go [hunk 2]
```diff
@@ -342,23 +361,24 @@ func (c *Impl) EnqueueNamespaceOf(obj interface{}) {
...
-// MaybeEnqueueBucketKey takes a Bucket and namespace/name string and puts it onto the work queue.
-func (c *Impl) MaybeEnqueueBucketKey(bkt reconciler.Bucket, key types.NamespacedName) {
+// MaybeEnqueueBucketKey takes a Bucket and namespace/name string and puts it onto
+// the slow work queue.
+func (c *Impl) maybeEnqueueBucketKey(bkt reconciler.Bucket, key types.NamespacedName) {
 	if bkt.Has(key) {
-		c.WorkQueue.Add(key)
-		c.logger.Debugf("Adding to queue %s (depth: %d)", safeKey(key), c.WorkQueue.Len())
+		c.workQueue.SlowLane().Add(key)
+		c.logger.Debugf("Adding to queue %s (depth: %d)", safeKey(key), c.workQueue.Len())
```
Comment (mattmoor) [inline controller/controller.go:373]: Call EnqueueSlowKey.

File: controller/controller.go [hunk 3]
```diff
@@ -342,23 +361,24 @@ func (c *Impl) EnqueueNamespaceOf(obj interface{}) {
...
 func (c *Impl) EnqueueKey(key types.NamespacedName) {
-	c.WorkQueue.Add(key)
-	c.logger.Debugf("Adding to queue %s (depth: %d)", safeKey(key), c.WorkQueue.Len())
+	c.workQueue.Add(key)
+	c.logger.Debugf("Adding to queue %s (depth: %d)", safeKey(key), c.workQueue.Len())
 }
 
-// MaybeEnqueueBucketKey takes a Bucket and namespace/name string and puts it onto the work queue.
-func (c *Impl) MaybeEnqueueBucketKey(bkt reconciler.Bucket, key types.NamespacedName) {
+// MaybeEnqueueBucketKey takes a Bucket and namespace/name string and puts it onto
```
Comment (mattmoor) [inline controller/controller.go:368]:
```suggestion
// maybeEnqueueBucketKey takes a Bucket and namespace/name string and puts it onto
```

File: controller/controller.go [hunk 4]
```diff
@@ -186,12 +185,14 @@ type Impl struct {
 	// from the workqueue to process.  Public for testing.
 	Reconciler Reconciler
 
-	// WorkQueue is a rate limited work queue. This is used to queue work to be
-	// processed instead of performing it as soon as a change happens. This
-	// means we can ensure we only process a fixed amount of resources at a
-	// time, and makes it easy to ensure we are never processing the same item
-	// simultaneously in two different workers.
-	WorkQueue workqueue.RateLimitingInterface
+	// workQueue is a rate limited two land work queue.
```
Comment (whaught) [inline controller/controller.go:188]:
```suggestion
	// workQueue is a rate-limited, two-lane work queue.
```

File: controller/controller.go (hunk 2 above)
Comment (vagababov) [reply controller/controller.go:373]: renamed but kept both in one. No usage of the other one so far.

File: controller/controller.go (hunk 3 above)
Comment (vagababov) [reply controller/controller.go:368]: made back publick.

File: controller/controller.go (hunk 4 above)
Comment (vagababov) [reply controller/controller.go:188]: Done, thanks.

**Comment by vagababov** [comment]
fixed merge and issues

**Comment by vagababov** [comment]
I'll deal with test separately
/retest

**Pull request merged**